### ISSUE TITLE
Create general ugly red warning start page for old versions

### ISSUE THREAD

**Issue by Sancus** [thundernest/thunderbird-website#170]
Right now we have the earlybird warning page: https://start.thunderbird.net/en-US/earlybird/

It's not localized, and we should probably change the copy to not talk about Earlybird:

"This version is no longer a supported release, and does not receive security updates!"
"We **strongly recommend** downloading the latest stable version of Thunderbird."

After localization, we should redirect start pages to it. I suggest everything below Thunderbird 52(So 45, 38, and everything below that would be the major versions affected). This is roughly 4% of our users.

**Comment by Sancus** [comment]
@ryanleesipes @wsmwk Any feedback on this copy or do you guys think it's good?

Test page here: https://start-stage.thunderbird.net/en-US/outdated/

**Comment by mkmelin** [comment]
I wouldn't mention beta there. These are the people who haven't upgraded at all for ages - they clearly don't want beta software. 
I think it could be good to also be a bit more specific. It seems a bit general now. Could be perhaps with JavaScript show an estimate of for how long they have been out of date, and a very rough estimate of how many security fixes they have missed out on?

**Comment by Sancus** [comment]
Yeah that's fair, I can remove beta.

It's general because it's meant to apply for ANY version below 52. So you can't make very many assumptions about what version it's displaying inside or anything like that.

I'm not sure of the benefit of adding Javascript to this page. It needs to at least not throw errors/not look broken on all versions of Thunderbird going back to 1.0...and I'm definitely not testing it on 20+ versions of Thunderbird lol.

Do you really think that adding that kind of text will persuade more people? I'm skeptical -- I don't think people will read or care about the number of missed security updates, and the people who understand what that means probably have already updated.

**Comment by mkmelin** [comment]
It's hard to say, but it's kind of common for software to tell you to update, so I'd want them to understand, they are now *really* out of date.

That, or we put a sad cat picture there, and if they click it they go to download :)

**Comment by wsmwk** [comment]
I have some feedback on this.  I'll try to post in the next few days

**Comment by wsmwk** [comment]
Sorry for the long delay.

As much as I hate losing potential beta users, I agree with Magnus about removing the beta link but not because "they clearly don't want beta software" which I don't see how we can assume just because they are uneducated about their version.  But rather because it simplifies the process, and getting them to a supported version with fewer support issues is more important than having them on beta, and since they haven't been on valid beta for such a long time it's not like we are losing testers.

As for timing (for beta and release),  I'd prefer to hold off several months until version 78 takes solid hold and our add-ons story is in much better shape - no sense giving them a jarring update to 68 (and getting used to that version) followed by another jarring update to an 78. 

Can we target nightly channel users now see how it goes?

**Comment by Sancus** [comment]
> As for timing (for beta and release), I'd prefer to hold off several months until version 78 takes solid hold and our add-ons story is in much better shape - no sense giving them a jarring update to 68 (and getting used to that version) followed by another jarring update to an 78.

Not totally sure I agree this is a good idea. As far as I understand, there are probably many add-ons that will never(or at least not anytime soon) see 78 versions. I would rather have people on 52 or earlier on 68 than anything, wouldn't you?

> 
> Can we target nightly channel users now see how it goes?

I'm not sure what this means? Wouldn't displaying this start page in Nightly be confusing? Do you mean people on really old versions of Nightly? I don't think there actually ARE any of those at all...

**Comment by wsmwk** [comment]
On 6/8/20 2:07 PM, Andrei Hajdukewycz wrote:
>
>     As for timing (for beta and release), I'd prefer to hold off
>     several months until version 78 takes solid hold and our add-ons
>     story is in much better shape - no sense giving them a jarring
>     update to 68 (and getting used to that version) followed by
>     another jarring update to an 78.
>
> Not totally sure I agree this is a good idea. As far as I understand, 
> there are probably many add-ons that will never(or at least not 
> anytime soon) see 78 versions. I would rather have people on 52 or 
> earlier on 68 than anything, wouldn't you?
>
No, absolutely I wouldn't.  They've already been on an old version 
multiple years - a few months will not have a significant impact on 
their usage or security and would only satisfy our impatience.  (yes, 
I'm also axnoius to get them on a supported release.

Yes, of course many addons will never come back - those we cannot do 
anything about and are not the point.  It's the ones that will come back 
that matter.  And so at least we can save a few users the grief of being 
on 68 for a couple months were virtually no addons work. Then going to 
from pre-68 to 78 will be just one pain rather than two.


>     Can we target nightly channel users now see how it goes?
>
> I'm not sure what this means? Wouldn't displaying this start page in 
> Nightly be confusing? Do you mean people on really old versions of 
> Nightly? I don't think there actually ARE any of those at all...
>
I don't understand.  Why would it be confusing if only users with old 
releases sees it?

Yes, there certainly are some 
https://crash-stats.mozilla.org/signature/?release_channel=nightly&product=Thunderbird&date=%3E%3D2019-12-08T18%3A17%3A00.000Z&date=%3C2020-06-08T18%3A17%3A00.000Z&_sort=-date&signature=OOM%20%7C%20small 
implies 5-10%, though I suspect it is less.  Do you have a number?

**Comment by Sancus** [comment]
> No, absolutely I wouldn't.  They've already been on an old version multiple years - a few months will not have a significant impact on their usage or security and would only satisfy our impatience.  

OK, I'm just thinking that it's going to be less successful trying to transition people directly to 78. I also think there are profile issues upgrading from 45 or 52 -> 78. Not totally sure. Magnus indicated that there are some things(password db?) that won't be migrated if you do this. We should test this and probably have a support article on issues you might run into doing this, tbh.

I agree in general getting people to a supported release is best, but there are quite a few security issues fixed as well even from 45 to 68, so it's not like there's no benefit there either...

> I don't understand.  Why would it be confusing if only users with old releases sees it? Yes, there certainly are some https://crash-stats.mozilla.org/signature/?release_channel=nightly&product=Thunderbird&date=%3E%3D2019-12-08T18%3A17%3A00.000Z&date=%3C2020-06-08T18%3A17%3A00.000Z&_sort=-date&signature=OOM%20%7C%20small implies 5-10%, though I suspect it is less.  Do you have a number?

I will have to check, the number of people on old Nightly versions is so small that it gets filtered out of the stats graphs as noise, but it is in the JSON files.

**Comment by wsmwk** [comment]
On 6/8/20 2:44 PM, Andrei Hajdukewycz wrote:
>
> OK, I'm just thinking that it's going to be less successful trying to 
> transition people directly to 78. I also think there are profile 
> issues upgrading from 45 or 52 -> 78. Not totally sure. Magnus 
> indicated that there are some things(password db?) that won't be 
> migrated if you do this. We should test this and probably have a 
> support article on issues you might run into doing this, tbh.
>
A valid thought.  But any migration items available in 68 I would hope 
will still work in newer versions than 68.

**Comment by mkmelin** [comment]
Nope, not all of them. That's why there are watersheds. m-c cleared out the old key3.db file support for instance (thus pwds from older installs would have to re-enter passwords) for instance.

**Comment by wsmwk** [comment]
On 6/8/20 4:18 PM, Magnus Melin wrote:
>
> Nope, not all of them. That's why there are watersheds. m-c cleared 
> out the old key3.db file support for instance (thus pwds from older 
> installs would have to re-enter passwords) for instance.
>
Yes, important to know the watersheds.  So

a) in the obsolete  warning we would want to tell users to expect 
multiple updates

b) do we have them download version 60, or version 68, or 78? (sorry, I 
don't remember all the  watersheds we've done, and the reasons)

Even if they have to go through watersheds, my opinion still holds that 
we shouldn't update them now to 68 only to push them in a couple months 
or weeks into another new version, 78.  IMO it should be a short quick 
path to a stable 78 with at least some add-ons - and that's not 78.0 and 
perhaps not even 78.1.

We also need to take into account users on hardware or OS that we no 
longer support.

**Comment by Sancus** [comment]
fyi there are roughly ~190 people on Nightly versions prior to 71.0a1 on any given week day. If you include 71.0a1 - 77.0a1 it's another few hundred. Total is around 4500 right now. Normal is more like 6-7k though, we're down like 30-40% since the Windows startup crash.

A couple of hundred people is probably enough to run a trial and see if we get any movement. We'll definitely have to take it with a grain of salt, especially since those users are probably quite different from regular release-build users. But since it's a simple thing to do, sure lets do it.

**Comment by Sancus** [comment]
https://start-stage.thunderbird.net/en-US/outdated/ I've updated this page. The version recognition works in Firefox/Thunderbird going back to at least 3.1, I tested. There is now a sentence mentioning a support article as well, though no article yet.

**Comment by Sancus** [comment]
On a weekday basis we have roughly the following users: 
45.*: <1K
52.*: 50K
60.*: 90K
68.*: 80K
78.*: 80K
91.*: 70K

Given that, we'll target 60 for the KB article about upgrading and people on versions prior to that will be on their own. They will still get the red start page, though.

**Comment by rtanglao** [comment]
* KB article is up to date: U[pgrading from an older version of Thunderbird](https://support.mozilla.org/en-US/kb/upgrading-older-version-thunderbird). Please go ahead and link to it!
